### PR TITLE
refactor(permalink): extract permalink parsing into its own module

### PR DESCRIPTION
**Pull request by Geobert**
To be used in pagination

### CODE REVIEW

File: src/cobalt_model/permalink.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+use liquid;
+use std::path::{Path, PathBuf};
+
+use error::*;
+
+pub fn explode_permalink<S: AsRef<str>>(
```
Comment (epage) [inline src/cobalt_model/permalink.rs]: Eventually we can consider refactoring this into a newtype so we can directly reference it in the frontmatter, etc and have everything work out.

File: src/document.rs [hunk 2]
```diff
@@ -13,6 +13,7 @@ use rss;
 
 use cobalt_model;
 use cobalt_model::files;
+use cobalt_model::permalink::{explode_permalink, format_url_as_file};
```
Comment (epage) [inline src/document.rs]: Could you switch to `use cobalt_model::permalink`?

First, I prefer that for the sake of making it easier to see where the code is coming from.

Second, this masked the fact that certain tests weren't moved over (github doesn't let me comment on code outside of the diff)

File: src/cobalt_model/permalink.rs (hunk 1 above)
Comment (Geobert) [reply src/cobalt_model/permalink.rs]: You mean, a proper `struct Permalink`?

File: src/document.rs (hunk 2 above)
Comment (Geobert) [reply src/document.rs]: Oh yeah, sorry I didn't noticed that it was not the adopted coding style, correcting this tomorrow :D (bed time here ^^')

Comment (epage) [reply src/document.rs]: I wasn't going to care enough but it caused us to miss the tests.

File: src/cobalt_model/permalink.rs (hunk 1 above)
Comment (epage) [reply src/cobalt_model/permalink.rs]: Yup.  It'll take some iterating on to figure out a design but I don't think its something you should worry about for this.

**Comment by Geobert** [comment]
I have a question on this one: if `permalink_attributes` in `document.rs` is not moved, how to I call it in the pagination code to process the `{{parent}}` and others already processed in `permalink_attributes`?

**Comment by epage** [comment]
> I have a question on this one: if permalink_attributes in document.rs is not moved, how to I call it in the pagination code to process the {{parent}} and others already processed in permalink_attributes?

I'm trying to move to iteratively design the `cobalt_model` API.  Anything not in there is something I don't have a good path forward on.  I think we might want to split or have all of the pagination logic outside of cobalt_model.  Feel free to make `permalink_attributes` `pub(crate)` but only use it outside of `cobalt_model`.

**Comment by Geobert** [comment]
Ok for the `pub(crate)` (can't wait for edition 2018 so we can write `crate` instead). Atm I have `pagination.rs` in `cobalt_model`, should I move it to `src/` then?

**Comment by epage** [comment]
Anything related to config or forntmatter should stay in cobalt_model.  Probably move the processing, including attributes, out.

**Pull request merged**